plugins/hypervisors/kvm/src/test/java/org/apache/cloudstack/kvm/ha/KvmHaAgentClientTest.java

```diff
@@ -0,0 +1,278 @@
+        if (expectNull) {
+            Assert.assertNull(response);
+        } else {
+            Assert.assertEquals(mockedResponse, response);
+        }
+    }
+
+    @Test
+    public void retryHttpRequestTestHttpOk() throws IOException {
+        HttpResponse mockedResponse = mockResponse(HttpStatus.SC_OK, JSON_STRING_EXAMPLE_3VMs);
+        Mockito.doReturn(mockedResponse).when(kvmHaAgentClient).retryUntilGetsHttpResponse(Mockito.anyString(), Mockito.any(), Mockito.any());
+        HttpResponse result = kvmHaAgentClient.retryHttpRequest(EXPECTED_URL, HTTP_REQUEST_BASE, client);
+        Mockito.verify(kvmHaAgentClient, Mockito.times(1)).retryUntilGetsHttpResponse(Mockito.anyString(), Mockito.any(), Mockito.any());
+        Assert.assertEquals(mockedResponse, result);
+    }
+
+    @Test
+    public void retryUntilGetsHttpResponseTestOneIOException() throws IOException {
+        Mockito.when(client.execute(HTTP_REQUEST_BASE)).thenThrow(IOException.class).thenReturn(mockResponse(HttpStatus.SC_OK, JSON_STRING_EXAMPLE_3VMs));
+        HttpResponse result = kvmHaAgentClient.retryUntilGetsHttpResponse(EXPECTED_URL, HTTP_REQUEST_BASE, client);
+        Mockito.verify(client, Mockito.times(MAX_REQUEST_RETRIES)).execute(Mockito.any());
+        Assert.assertNotNull(result);
+    }
+
+    @Test
+    public void retryUntilGetsHttpResponseTestTwoIOException() throws IOException {
+        Mockito.when(client.execute(HTTP_REQUEST_BASE)).thenThrow(IOException.class).thenThrow(IOException.class);
+        HttpResponse result = kvmHaAgentClient.retryUntilGetsHttpResponse(EXPECTED_URL, HTTP_REQUEST_BASE, client);
+        Mockito.verify(client, Mockito.times(MAX_REQUEST_RETRIES)).execute(Mockito.any());
+        Assert.assertNull(result);
+    }
+
+    @Test
+    public void isKvmHaWebserviceEnabledTestDefault() {
+        Assert.assertFalse(kvmHaAgentClient.isKvmHaWebserviceEnabled());
+    }
+
+    @Test
+    public void getKvmHaMicroservicePortValueTestDefault() {
+        Assert.assertEquals(KVM_HA_WEBSERVICE_PORT, kvmHaAgentClient.getKvmHaMicroservicePortValue());
+    }
+
+//    private void prepareAndRunCountRunningVmsOnAgent(String jsonStringExample, int expectedListedVms) throws IOException {
+//        Mockito.when(agent.getPrivateIpAddress()).thenReturn(PRIVATE_IP_ADDRESS);
+//        Mockito.doReturn(mockResponse(HttpStatus.SC_OK, JSON_STRING_EXAMPLE_3VMs)).when(kvmHaAgentClient).executeHttpRequest(EXPECTED_URL);
+//
+//        JsonObject jObject = new JsonParser().parse(jsonStringExample).getAsJsonObject();
+//        Mockito.doReturn(jObject).when(kvmHaAgentClient).processHttpResponseIntoJson(Mockito.any(HttpResponse.class));
+//
+//        int result = kvmHaAgentClient.countRunningVmsOnAgent();
+//        Assert.assertEquals(expectedListedVms, result);
+//    }
+//TODO
+//    @Test
+//    public void isTargetHostReachableTest() {
+//        kvmHaAgentClient.isTargetHostReachable(PRIVATE_IP_ADDRESS);
+//    }
```

Comment:
Do we need commented code?

Reply:
@GutoVeronezi No, we don't. Sorry, forgot to convert for a draft. I am adding some checks to enhance the HA validation.

The idea here is to verify if the Suspect Host is reachable by other hosts on the cluster, therefore excluding network issues between mgmt and kvm agents and avoiding fencing / rebooting in such cases.